plugins/genesys-gl32xx/fu-genesys-gl32xx-device.c

```diff
@@ -912,11 +918,38 @@ fu_genesys_gl32xx_device_set_progress(FuDevice *self, FuProgress *progress)
 	fu_progress_add_step(progress, FWUPD_STATUS_DEVICE_BUSY, 45, "reload");
 }
 
+static gboolean
+fu_genesys_gl32xx_device_set_quirk_kv(FuDevice *device, const gchar *key, const gchar *value, GError **error)
+{
+	FuGenesysGl32xxDevice *self = FU_GENESYS_GL32XX_DEVICE(device);
+	guint64 tmp;
+
+	/* load from quirks */
+	if (g_strcmp0(key, "CompatibleModel") == 0) {
+		if (!fu_strtoull(value, &tmp, 0, G_MAXUINT16, error))
+			return FALSE;
+
+		self->compatible_model = tmp;
+		return TRUE;
+	}
+
+	/* failed */
+	g_set_error_literal(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "quirk key not supported");
+	return FALSE;
+}
+
+
 static void
 fu_genesys_gl32xx_device_init(FuGenesysGl32xxDevice *self)
 {
 	self->packetsz = 64;
-	fu_device_set_vendor(FU_DEVICE(self), "Genesys");
+	self->compatible_model = 0;
+	if (fu_device_has_vendor_id(FU_DEVICE(self), "USB:05E3")) {
+		fu_device_set_vendor(FU_DEVICE(self), "Genesys");
+	}
```

Comment:
this you can do with:

    [USB\VID_05E3]
    Vendor = Genesys

Reply:
In the quirk file? Not something like `[BLOCK\VEN_05E3]` ? What's the difference between those formats?

But actually we might be able to use the default name fwupd gets, let me check I think we have a firmware that makes it looks like a generic genesys device.

Reply:
This is what it looks like without quirk:

```
> sudo ./venv/bin/fwupdtool get-devices --plugins genesys_gl32xx
Place your right index finger on the fingerprint reader
Loading…                 [************************************** ]
WARNING: This package has not been validated, it may not work properly.
Framework Laptop (12th Gen Intel Core)
│
└─MassStorageClass:
      Device ID:          7818afbec6007ce4fb6e2792851f6e46eefb7d85
      Current version:    2907
      Vendor:             Generic (BLOCK:0x05E3)
      GUID:               0cc25308-a257-5fe9-8183-abad7b85b050 ← BLOCK\VEN_05E3&DEV_0764&VER_29&CID_E4F090C2
      Device Flags:       • Updatable
                          • Is in bootloader mode
                          • Cryptographic hash verification is available
                          • Unsigned Payload
```

Coming from the SCSI strings:

```
> lsscsi --no-nvme
[1:0:0:0]    disk    Generic  MassStorageClass 2907  /dev/sdb 
```

For reference, the USB descriptor looks like this:

```
> lsusb -d 05e3:
Bus 002 Device 017: ID 05e3:0764 Genesys Logic, Inc. USB Storage
```

So we should still do the override the name.